### PR TITLE
Bugfix for requirements.txt options ordering

### LINKED ISSUE #236: 4.2 re-orders requirements.txt before installing, breaking extra pypi index
We have some packages in a secondary pypi server so our requirements.txt looks something like:

```
-i https://pypi.org/simple
--extra-index-url https://repo.fury.io/techadmin/
amazon-kclpy==1.5.0
argh==0.26.2
...
```

A recent change this package to `pip.js` seems to re-order this:
https://github.com/UnitedIncome/serverless-python-requirements/blob/master/lib/pip.js#L287

to
```
--extra-index-url https://repo.fury.io/techadmin/
-i https://pypi.org/simple
amazon-kclpy==1.5.0
argh==0.26.2
```

Now, this works fine with pip install, but when serverless tries to install with this file it can't find anything in the repo.fury.io index.

### PR DESCRIPTION
**Pull request by AndrewFarley**
Fixes #236

@vickeryj Can you please try this?

```
rm -Rf node_modules/serverless-python-requirements
npm i github:andrewfarley/serverless-python-requirements#bug-fix-requirements-ordering
```

### CODE REVIEW

File: lib/pip.js [hunk 1]
```diff
@@ -286,7 +286,7 @@ function generateRequirementsFile(source, target, options) {
       return false;
     }
     // If we have requirements.txt options (prefixed with --) keep them
-    if (req.length > 1 && req[0] == '-' && req[1] == '-') {
+    if (req.length > 1 && req.startsWith('--')) {
```
Comment (dschep) [inline lib/pip.js]: you can also remove the length check. `''.startsWith('--') === false`

Comment (AndrewFarley) [reply lib/pip.js]: Done, and done to the above check as well

**Comment by dschep** [comment]
Released as v4.2.1

**Comment by bryantbiggs** [comment]
@dschep it looks like CircleCI starting failing starting with this PR

**Comment by dschep** [comment]
Yeah I realized this. See #239 for fixes. I don't think it's actually the result of a bug.

**Pull request merged**